src/libstd/sys/common/net.rs

```diff
@@ -221,11 +226,12 @@ impl TcpStream {
 
     pub fn write(&self, buf: &[u8]) -> io::Result<usize> {
         let len = cmp::min(buf.len(), <wrlen_t>::max_value() as usize) as wrlen_t;
+        let flags = if cfg!(target_os = "linux") { MSG_NOSIGNAL } else { 0 };
```

Comment:
Yes indeed. I was was just trying to emulate https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/net.rs#L26 .

You prefer to pass `MSG_NOSIGNAL` all the time ? I guess we can dispense with the `let flags` and put it straight in the send call in that case.